snewpdag/plugins/ValidateListType.py

```diff
@@ -0,0 +1,69 @@
+'''
+This will check the types of the elements in a list,
+and will consume the action if
+more than 10% of the data are not the right type.
+Then we don't have to have each downstream plugin
+repeat this sort of validation.
+
+check_listtype() checks the type of the elements in the list
+based on the type we want in the input,
+and if a wrong type is found, such element is removed
+and the removal is logged.
+
+min_fraction is the maximum fraction of numbers with wrong key_type allowed
+that we don't need to consume the action.
+e.g. min_fraction = 0.1 (i.e. 10%)
+I. if there are 10% or fewer elements in the list with wrong key type,
+we simply remove the element and log the removal.
+II. if there are more than 10% with wrong key type, we consume the action.
+
+Input: data as payload + min_fraction + key_type
+'''
+
+import logging
+from snewpdag.dag import Node
+
+class ValidateListType(Node):
+    def __init__(self, max_fraction, **kwargs):
+        self.max_fraction = max_fraction
+        self.key_type = kwargs.pop('key_type', None)
+        super().__init__(**kwargs)
+    
+    def check_listtype(self, data): # same as above but we check elements of a list
+        data_len = len(data)
+        remove_element = []
+        for x in data:
+            if type(x).__name__ != self.key_type:
+                remove_element.append(x)
+        if len(remove_element) <= (data_len * self.max_fraction):
+            logging.error('{} elements in list are not the desired type of {} and are deleted'.format(len(remove_element), self.key_type))
+            for x in remove_element:
+                data.remove(x)
+            return data
+        else:
+            logging.error('More than 10% in list are not the desired type of {} and action is consumed'.format(self.key_type))
+            return False
+    
+    def alert(self, data):
```

Comment:
these methods can be more concise, as you did in ValidateKey

Reply:
I wrote it as 
      if self.check_listtype(data) != False:
            return True
      else:
            return False
because self.check_listtype(data) may return the updated data array itself, instead of a pure True/False boolean, so I'm not sure if there is a cleaner way to do this.